packages/client/foreman_ygg_worker/foreman_ygg_worker.spec

```diff
@@ -0,0 +1,37 @@
+%global repo_orgname adamruzicka
+%global repo_name foreman_ygg_worker
+%global yggdrasil_libexecdir %{_libexecdir}/yggdrasil
+
+Name: foreman_ygg_worker
+Summary: Worker service for yggdrasil that can act as pull client for Foreman Remote Execution
+Version: 0.0.1
+Release: 1%{?dist}
+License: MIT
+
+Source0: https://github.com/%{repo_orgname}/%{repo_name}/releases/download/v%{version}/%{repo_name}-linux-amd64.tar.gz
```

Comment:
Does this contain all dependencies? I was comparing this spec file to what `go2pm` generates, and I noticed that it generates explicit dependencies as BuildRequires:

```
BuildRequires:  golang(git.sr.ht/~spc/go-log)
BuildRequires:  golang(github.com/google/uuid)
BuildRequires:  golang(github.com/redhatinsights/yggdrasil/protocol)
BuildRequires:  golang(google.golang.org/grpc)
```

Since it seems to do that build in the spec file:

```
%description
%{common_description}

%gopkg

%prep
%goprep

%build
for cmd in src; do
  %gobuild -o %{gobuilddir}/bin/$(basename $cmd) %{goipath}/$cmd
done

%install
%gopkginstall
install -m 0755 -vd                     %{buildroot}%{_bindir}
install -m 0755 -vp %{gobuilddir}/bin/* %{buildroot}%{_bindir}/
```

Reply:
TIL `go2rpm` exists. I tried putting together a spec by hand which didn't go exactly great. I'll give it another shot.

Reply:
The worker itself requires `git.sr.ht/~spc/go-log`, `github.com/google/uuid`, `github.com/redhatinsights/yggdrasil` and `google.golang.org/grpc`.

`git.sr.ht/~spc/go-log` and `github.com/google/uuid` have no dependencies, `google.golang.org/grpc` seems to be packaged (or at least in fedora's repos), however yggdrasil itself has quite a few dependencies (see https://github.com/RedHatInsights/yggdrasil/blob/main/go.mod). There might be overlap between yggdrassil's and grpc's dependencies, but it would still mean we would have to carry a lot of random build dependencies in our repos. Is that really what we want?

Reply:
Not if I can help it, however, I am trying to make sure we really think through it since that has bit us with nodejs sometimes. And, building binaries outside of the RPM can get weird sometimes.

Is it possible to bundle the dependencies but still build it inside the RPM environment? Similar question if providing a binary as the source makes things easier or harder for Debian. @evgeni thoughts?

Reply:
I always thought that in Go land, it's usual to vendor stuff inside your app. We do have one Go think already packaged: [pcp-mmvstatsd](https://github.com/theforeman/foreman-packaging/tree/rpm/develop/packages/foreman/pcp-mmvstatsd) and it uses a `vendor/` folder with its deps, so we only fetch one tarball and then go `go build` (SCNR):

https://github.com/theforeman/foreman-packaging/blob/331d9e9058e8da9c5504d6f52aede35fc346242d/packages/foreman/pcp-mmvstatsd/pcp-mmvstatsd.spec#L30-L38

I have no idea about the Debian side of things.